ros_buildfarm/templates/release/rpm/sync_packages_to_testing_job.xml.em

```diff
@@ -96,62 +96,13 @@
-        'python3 -u $WORKSPACE/ros_buildfarm/scripts/release/rpm/list_packages.py' +
-        ' --pulp-base-url http://repo:24817' +
-        ' --pulp-distribution-name ros-building-%s-%s-%s' % (os_name, os_code_name, arch) +
-        ' --package-name-expression "^ros-%s-.*"' % rosdistro_name +
-        ' --pulp-resource-record $WORKSPACE/ros-building-binary.txt',
-        'echo "# END SECTION"',
-        '',
-        'echo "# BEGIN SECTION: determine binary debug packages to sync"',
-        'export PYTHONPATH=$WORKSPACE/ros_buildfarm:$PYTHONPATH',
-        'python3 -u $WORKSPACE/ros_buildfarm/scripts/release/rpm/list_packages.py' +
-        ' --pulp-base-url http://repo:24817' +
-        ' --pulp-distribution-name ros-building-%s-%s-%s-debug' % (os_name, os_code_name, arch) +
-        ' --package-name-expression "^ros-%s-.*"' % rosdistro_name +
-        ' --pulp-resource-record $WORKSPACE/ros-building-binary-debug.txt',
-        'echo "# END SECTION"',
-    ]),
-))@
-@(SNIPPET(
-    'builder_shell',
-    script='\n'.join([
-        'echo "# BEGIN SECTION: sync source packages to testing repo"',
-        'export PYTHONPATH=$WORKSPACE/ros_buildfarm:$PYTHONPATH',
-        'sed "s/^PULP_RESOURCES=//" $WORKSPACE/ros-building-source.txt | xargs' +
-        ' python3 -u $WORKSPACE/ros_buildfarm/scripts/release/rpm/import_package.py' +
-        ' --pulp-base-url http://repo:24817' +
-        ' --pulp-distribution-name ros-testing-%s-%s-SRPMS' % (os_name, os_code_name) +
-        ' --invalidate-expression "^ros-%s-.*"' % rosdistro_name,
-        'echo "# END SECTION"',
-        '',
-        'echo "# BEGIN SECTION: sync binary packages to testing repo"',
-        'export PYTHONPATH=$WORKSPACE/ros_buildfarm:$PYTHONPATH',
-        'sed "s/^PULP_RESOURCES=//" $WORKSPACE/ros-building-binary.txt | xargs' +
-        ' python3 -u $WORKSPACE/ros_buildfarm/scripts/release/rpm/import_package.py' +
-        ' --pulp-base-url http://repo:24817' +
-        ' --pulp-distribution-name ros-testing-%s-%s-%s' % (os_name, os_code_name, arch) +
-        ' --invalidate-expression "^ros-%s-.*"' % rosdistro_name,
-        'echo "# END SECTION"',
-        '',
-        'echo "# BEGIN SECTION: sync binary debug packages to testing repo"',
-        'export PYTHONPATH=$WORKSPACE/ros_buildfarm:$PYTHONPATH',
-        'sed "s/^PULP_RESOURCES=//" $WORKSPACE/ros-building-binary-debug.txt | xargs' +
-        ' python3 -u $WORKSPACE/ros_buildfarm/scripts/release/rpm/import_package.py' +
-        ' --pulp-base-url http://repo:24817' +
-        ' --pulp-distribution-name ros-testing-%s-%s-%s-debug' % (os_name, os_code_name, arch) +
-        ' --invalidate-expression "^ros-%s-.*"' % rosdistro_name,
+        ' --distribution-source-expression "^ros-building-%s-%s-(SRPMS|%s(-debug)?)$"' % (os_name, os_code_name, arch) +
+        ' --distribution-dest-expression "ros-testing-%s-%s-\\1"' % (os_name, os_code_name) +
+        ' --package-name-expression "^ros-%s-.*"' % rosdistro_name,
```

Comment:
The arguments to `sync_repo.py` might make more sense after looking at these arguments.

I'm using a regular expression replacement to transform matching source distributions to a destination distribution, then matching packages within the destination are removed and matching packages in the source are added to the destination.